src/neuroconv/utils/json_schema.py

```diff
@@ -48,6 +53,64 @@ def get_base_schema(tag=None, root=False, id_=None, **kwargs) -> dict:
     return base_schema
 
 
+def get_pydantic_model_from_method_signature(function: callable, exclude: Optional[List[str]] = None) -> BaseModel:
+    """
+    Generate a Pydantic model from the signature of an arbitrary function which uses keywords with annotation typing.
+
+    Parameters
+    ----------
+    function : callable
+        Any function that uses keywords with annotation types.
+    exclude : list of strings, default: None
+        A list of keys to exclude from the model but are present in the function signature.
+        Can be useful when skipping reserved names such as 'self' in methods of a class.
+
+    Returns
+    -------
+    model : instance of a pydantic model
+    """
+    # The default model of the parent 'ValidatedFunction' includes many fields we do not need
+    # but typical model-copying methods only work on fully instantiated models (which also triggers validation),
+    # whereas the model of this decorator class works more as a class method
+    class CustomValidatedFunction(ValidatedFunction):
```

Comment:
I will say, due to the underdocumented nature of this function on Pydantic, I don't actually know if it's intended to be used generally. It seems to still be undergoing active development, and many values we might want to control (including more detailed `configs`) don't seem to be exposed or propagate very well.

Just something to keep in mind if we decide we want to offload all this to another library